clang/docs/ReleaseNotes.rst

```diff
@@ -102,6 +102,16 @@ Deprecated Compiler Flags
 Modified Compiler Flags
 -----------------------
 
+- The ``-ffp-model`` option has been updated to enable a more limited set of
+  optimizations when the ``fast`` argument is used and to accept a new argument,
+  ``aggressive``. The behavior of ``-ffp-model=aggressive`` is equivalent
+  to the previous behavior of ``-ffp-model=fast``. The updated
+  ``-ffp-model=fast`` behavior no longer assumes finite math only, uses
+  the ``promoted`` algorithm for complex division when possible rather than the
+  less basic (limited range) algorithm, and sets
+  ``-ffp-contract=fast-honor-pragmas`` rather than ``-ffp-contract=fast``.
```

Comment:
I'm not super comfortable with this approach because it's a silent behavior change between compiler releases; would it be reasonable to deprecate "fast" entirely (leaving its behavior alone) and introduce two new options: `aggressive` and `user-hostile` (or something perhaps less loaded, lol)?

Reply:
Err, that's not building my confidence in the current approach. The justifications for why it's reasonable are defensible, but the reality still is that any significant change in behavior between compiler upgrades will cause friction for users and this has observable differences in numerical results and possibly performance. Silent changes like that can sometimes come across as user-hostile for users who hit that friction.

That said, I'd appreciate some additional opinions because it's possible I'm being too conservative. CC @jcranmer-intel @MaskRay @jyknight

Reply:
FWIW, @andykaylor and I spoke about my concerns offline and I'm less concerned now. The silent behavior change is actually protecting the user from more aggressive optimizations they may not have intended to apply to their code anyway, so while it is a silent behavioral change, it's likely to be the better behavior for the most users, even if it's worse for a few users.